 __tests__/pet.test.js

```diff
@@ -0,0 +1,247 @@
+  it("increments the age by 1", () => {
+    const pet = new Pet("Dougal");
+
+    pet.growUp();
+
+    expect(pet.age).toEqual(1);
+  });
+
+  it("increments the hunger by 5", () => {
+    const pet = new Pet("Dougal");
+
+    pet.growUp();
+
+    expect(pet.hunger).toEqual(5);
+  });
+
+  it("decreases the fitness by 3", () => {
+    const pet = new Pet("Dougal");
+
+    pet.growUp();
+
+    expect(pet.fitness).toEqual(7);
+  });
+
+  it("throws an error if the pet is not alive", () => {
+    const pet = new Pet("Dougal");
+
+    pet.hunger = 12;
+
+    expect(() => pet.growUp()).toThrow("Your pet is no longer alive :(");
+  });
+});
+
+describe("walk", () => {
+  it("increases fitness by to a maximum of 10", () => {
+    const pet = new Pet("Toto");
+
+    pet.fitness = 8;
+    pet.walk();
+
+    expect(pet.fitness).toEqual(10);
+  });
+
+  it("throws an error if the pet is not alive", () => {
+    const pet = new Pet("Toto");
+
+    pet.fitness = 0;
+
+    expect(() => pet.walk()).toThrow("Your pet is no longer alive :(");
+  });
+});
+
+describe("feed", () => {
+  it("decreases the hunger by 3", () => {
+    const pet = new Pet("Brian");
+
+    pet.hunger = 8;
+    pet.feed();
+
+    expect(pet.hunger).toEqual(5);
+  });
+
+  it("hunger level should not go below 0", () => {
+    const pet = new Pet("Brian");
+
+    pet.hunger = 1;
+    pet.feed();
+
+    expect(pet.hunger).toEqual(0);
+  });
+
+  it("throws an error if the pet is not alive", () => {
+    const pet = new Pet("Brian");
+
+    pet.age = 30;
+
+    expect(() => pet.feed()).toThrow("Your pet is no longer alive :(");
+  });
+});
+
+describe("checkUp", () => {
+  it("checks if the fitness is 3 or less", () => {
+    const pet = new Pet("Lassie");
+
+    pet.fitness = 3;
+    pet.checkUp();
+
+    expect(pet.checkUp()).toBe("I need a walk");
+  });
+
+  it("checks if the hunger is 5 or more", () => {
+    const pet = new Pet("Lassie");
+
+    pet.hunger = 7;
+    pet.checkUp();
+
+    expect(pet.checkUp()).toBe("I am hungry");
+  });
+
+  it("checks if the fitness is 3 or less and if hunger is 5 or more", () => {
+    const pet = new Pet("Lassie");
+
+    pet.fitness = 2;
+    pet.hunger = 6;
+    pet.checkUp();
+
+    expect(pet.checkUp()).toBe("I am hungry AND I need a walk");
+  });
+
+  it("checks if the fitness is 4 or more and if hunger is 4 or less", () => {
+    const pet = new Pet("Lassie");
+
+    pet.fitness = 8;
+    pet.hunger = 2;
+    pet.checkUp();
+
+    expect(pet.checkUp()).toBe("I feel great!");
+  });
+
+  it("throws an error if the pet is not alive", () => {
+    const pet = new Pet("Lassie");
+
+    pet.age = 42;
+
+    expect(() => pet.checkUp()).toThrow("Your pet is no longer alive :(");
+  });
+});
+
+describe("have Baby", () => {
+  it("allows the Pet to have children", () => {
+    const parent = new Pet("Nina");
+
+    parent.haveBaby("Maru");
+
+    expect(parent.haveBaby("Maru")).toEqual(
+      "Congratulations Nina on your child Maru!"
+    );
```

Comment:
Since youre making this same function call twice, can you be sure that you havent just created 2 children? A way to check this would be to assert the length of the children array on the parent:
```
    expect(parent.children.length).toBe(1);
```
Try adding this assertion here, does the test fail or pass? If it fails, you should remove the function on line 231 and the problem should be fixed